l1-contracts/src/core/Rollup.sol

```diff
@@ -95,9 +95,9 @@ contract Rollup is IRollup {
       header.globalVariables.blockNumber, header.contentCommitment.outHash, l2ToL1TreeHeight
     );
 
-    // pay the coinbase 1 gas token if it is not empty
-    if (header.globalVariables.coinbase != address(0)) {
-      GAS_TOKEN.transfer(address(header.globalVariables.coinbase), 1);
+    // pay the coinbase 1 gas token if it is not empty and header.totalFees is not zero
+    if (header.globalVariables.coinbase != address(0) && header.totalFees > 0) {
```

Comment:
Should we pay if `totalFees` is not zero and throw if coinbase is empty?

Reply:
Yes, and there are comments around the codebase saying that coinbase should not be allowed to be undefined when we are in production. Not all the tests set coinbase yet. So I added #6539

Reply:
Cool! The rest looks good to me 😃